src/core/namespace.c

```diff
@@ -658,9 +674,17 @@ static int mount_private_dev(MountEntry *m) {
                         goto fail;
                 }
         } else {
-                r = clone_device_node("/dev/ptmx", temporary_mount);
-                if (r < 0)
-                        goto fail;
+                r = clone_device_node("/dev/ptmx", temporary_mount, can_mknod);
+                if (r != -EPERM) {
+                        if (errno != EPERM)
+                                goto fail;
```

Comment:
we generally return errors as negative return values, hence no need to check errno here. clone_device_node() should have converted that into a return value already.

Reply:
addressed